src/convert.jl

```diff
@@ -22,3 +22,40 @@ convert(::Type{NamedVector}, a::AbstractArray) = NamedArray(a)
 convert(::Type{NamedArray{T}}, n::NamedArray) where {T} = NamedArray(T.(n.array), n.dicts, n.dimnames)
 
 @inline convert(::Type{NamedArray}, n::NamedArray) = n
+
+#= """ =#
+#= Conversion from a Dict to a NamedArray. =#
+#= """ =#
+#= function convert(t::Type{NamedArray}, dict::Dict{K,V}) where {K,V} =#
+#= 	allkeys = collect(keys(dict)) =#
+#= 	allvalues = collect(values(dict)) =#
+#= 	return NamedArray(allvalues, allkeys, (:keys)) =#
+#= end =#
+
+using DataFrames
+"""
+Converts a NamedArray to a DataFrame.
+
+"""
+function convert(t::Type{DataFrame}, n::NamedArray{V}; valueCol = :Values) where {V}
+	mydimnames = n.dimnames
+	mytypes = map(dict->eltype(keys(dict)),n.dicts)
+	dfArgs = map((d,t)-> d=>t[], mydimnames, mytypes)
+	dfArgs = (dfArgs..., valueCol => V[])
+	df = DataFrame(dfArgs...)
+	map(tup -> push!(df,(tup[1]...,tup[2])),enamerate(n))
+	return df
+end
+
+"""
+Converts a DataFrame to a NamedArray.
+"""
+function convert(t::Type{NamedArray}, df::DataFrame; valueCol = :Values)
+	newdimnames = propertynames(df)
+	deleteat!(newdimnames,findfirst(x->x==:Values,newdimnames))
```

Comment:
I think you want here :
`deleteat!(newdimnames,findfirst(x->x==valueCol,newdimnames))`
rather than:
`deleteat!(newdimnames,findfirst(x->x==:Values,newdimnames))`